### PR TITLE
[WIP] Upgrade to PHP 8.1 and Laminas Framework

### PR DESCRIPTION
**Pull request by tomudding**
This is is still WIP, however, I need the actions to do their job so I cannot make this an actual WIP PR.

Most of the things are done and in working order. However, a few things do not work:

- [x] Entity creation for most models, the hydrators interfere with the typed properties. I will likely remove the hydrators even though they are very useful.
- [x] If the above works, an issue exists in the `Foundation` form, which prevents the creation of organs. The `MemberFunctionFieldset` breaks validation with the following message*:

> No element by the name of [0] found in form.

- [ ] Member enrolment form does not work.
- [ ] Address lost when approving prospective member.

Changes in this PR:
- [X] Moved to PSR-4 autoloading.
- [X] Upgraded to Zend Framework 3
- [X] Remove old and unused `Export` and `Import` modules.
- [X] Upgraded to PHP 8.1 and Laminas Framework
- [X] Removed custom event logger
- [X] Switched to native attributes for Doctrine annotations
- [X] Fixed legacy code style issues.

*: The problem occurs when trying to call `isValid()` on the submitted form. Removing the `bind()` for `Decision` shows that the data is actually present, so something happens when binding the values which removes the data.

### CODE REVIEW

**Comment by tomudding** [comment]
I think the best solution for the hydrators is to remove them when creating entities, not when editing them. That way they are still useful but do not impact how we use static typing.

Regarding the broken form, some information on what I tried: I tried giving everything a factory and using `init()` for forms that contained fieldsets. That didn't work because of the dependency injections. Next I tried to use the `FormElementManager` to get proxies of the forms instead of actual instances, however, this broke other forms. I think I spend a good 8 hours trying all kinds of different things and nothing worked in the end.

**Comment by tomudding** [comment]
Good news, I have found a way to fix most if not all issues with the hydrators. `doctrine/doctrine-laminas-hydrator` has a special property for `DoctrineObject` which allows extracting and hydrating via reference and not the actual entity API.

**Comment by Koen1999** [comment]
@tomudding, you have done some amazing work!

I think you are very much on the right track here, although of course there still are some issues to fix.

I looked into the dependencies and noticed a small issue with Laminas/Console I fixed.

Overall, it is more than likely that there's some bugs.
To aid in this process, I added the automatic tests we also have in gewisweb.
Since PHP has become more strict, in the newer versions there's already some issues arising.

**Comment by Koen1999** [comment]
Moreover, I propose we merge this soon. Then you can work on a new PR on which the CI actions actually work. That may be useful :p

File: module/User/src/Module.php [hunk 1]
```diff
@@ -58,18 +52,6 @@ public function onBootstrap($event)
...
-            // Check if not a console route, as the console has no headers.
-            if (!($e->getResponse() instanceof Response)) {
-                if (!$authService->hasIdentity()) {
-                    $response = $e->getResponse();
-                    $response->getHeaders()->addHeaderLine('Location', '/user');
-                    $response->setStatusCode(302);
-                    return $response;
-                }
-            }
-        });
```
Comment (tomudding) [inline module/User/src/Module.php:71]: I do think that we can and perhaps should keep this, but without checking `$e->getResponse()` of course.

Comment (Koen1999) [reply module/User/src/Module.php]: That's also fine!

**Comment by tomudding** [comment]
I think I have now fixed all major issues (apart from enrolling, which I will do in a separate PR). I have also typed most if not all pieces of code (and tested almost all of it) and it is working great. I did have to add the same nasty hack for the installation form but at least it works again.

I will merge this now to actually get the actions working.

**Comment by tomudding** [comment]
I might have forgotten to rename this, but alas.

**Pull request merged**